src/core/input_descriptor.rs

```diff
@@ -272,12 +272,15 @@ impl Constraints {
 #[derive(Debug, Clone)]
 pub struct CompiledJsonSchema {
     raw: serde_json::Value,
-    compiled: Arc<jsonschema::JSONSchema>,
+    compiled: Arc<jsonschema::Resource>,
```

Comment:
Is this Arc still needed if the methods now don't return `self.compiled` directly?